app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -84,15 +87,51 @@ private void startPushService() {
             missingMessageUtil.lastReceivedMessage(lastReceivedMessage::set);
         }
 
+        ConnectivityManager cm =
+                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+
         connection =
-                new WebSocketConnection(settings.url(), settings.sslSettings(), settings.token())
+                new WebSocketConnection(
+                                settings.url(), settings.sslSettings(), settings.token(), cm)
                         .onOpen(this::onOpen)
                         .onClose(() -> foreground(getString(R.string.websocket_closed)))
                         .onBadRequest(this::onBadRequest)
-                        .onFailure((min) -> foreground(getString(R.string.websocket_failed, min)))
+                        .onReconnectSchedule(
```

Comment:
IMO a message for "reconnect scheduled" isn't needed, currently it would only be shown for five seconds, I rather have no switch case than this extra information. Also the method (onReconnectSchedule) can then be renamed to onFailure which also looks cleaner.

Reply:
Actually I renamed it in the first place because the name `onFailure` is somewhat confusing. I noticed that you added some logic on `Listener.onFailure` to route 4XX errors to `onBadRequest`. Thus I think here we should not use the `onFailure` name again as it actually only fires on connectivyty errors, and `onReconnectSchedule` is clear enough on how the service should react to this event.

Reply:
I partly agree with you there, but currently the `Connection failed, trying again in %d minutes` is shown on a scheduled reconnect this is right when the connection failed but not after the reconnect listener was triggered.

How about renaming onBadRequest to onUnresolvableFailure and onReconnectScheduled to onConnectionFailure?

Reply:
That sounds good